pkg/controller/notification/notification_controller.go

```diff
@@ -135,7 +162,8 @@ func (r *ReconcileNotification) checkTransitionTimeAndDelete(log logr.Logger, no
 	return false, diff, nil
 }
 
-func (r *ReconcileNotification) wrapErrorWithStatusUpdate(logger logr.Logger, notification *toolchainv1alpha1.Notification, statusUpdater func(notification *toolchainv1alpha1.Notification, message string) error, err error, format string, args ...interface{}) error {
+func (r *ReconcileNotification) wrapErrorWithStatusUpdate(logger logr.Logger, notification *toolchainv1alpha1.Notification,
+	statusUpdater func(notification *toolchainv1alpha1.Notification, message string) error, err error, format string, args ...interface{}) error {
```

Comment:
you could create a `statusUpdater` type like this:
```
type statusUpdater func(notification *toolchainv1alpha1.Notification, message string) error
```
and your func signature would become:
```
func (r *ReconcileNotification) wrapErrorWithStatusUpdate(logger logr.Logger, notification *toolchainv1alpha1.Notification, statusUpdater statusUpdater, err error, format string, args ...interface{}) error
```
which is a little simpler to read

Reply:
good idea! changed as part of: 057eb87e2123a2edfa90a49c007140b845d59010